public/app/plugins/panel/graph3/plugins/ExemplarsPlugin.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { map } from 'rxjs/operators';
+import { AnnotationEvent, DataFrame, dateTimeFormat, systemDateFormats, TimeZone } from '@grafana/data';
+import { getAnnotationsFromData } from 'app/features/annotations/standardAnnotationSupport';
+import { EventsCanvas, usePlotContext } from '@grafana/ui';
+import { ExemplarMarker } from './ExemplarMarker';
+import { Subscription } from 'rxjs';
+
+interface ExemplarsPluginProps {
+  exemplars: DataFrame[];
+  timeZone: TimeZone;
+}
+
+// tmp type to mock an exemplar events
+interface ExemplarEvent extends AnnotationEvent {
+  y: number;
+}
+
+export const ExemplarsPlugin: React.FC<ExemplarsPluginProps> = ({ exemplars, timeZone }) => {
+  const plotCtx = usePlotContext();
+  const [exemplarEvents, setExemplarEvents] = useState<ExemplarEvent[]>([]);
+
+  const timeFormatter = useCallback(
+    (value: number) => {
+      return dateTimeFormat(value, {
+        format: systemDateFormats.fullDate,
+        timeZone,
+      });
+    },
+    [timeZone]
+  );
+
+  useEffect(() => {
+    let subscription: Subscription;
+    if (plotCtx.isPlotReady) {
+      subscription = getAnnotationsFromData(exemplars)
+        .pipe(
+          map<AnnotationEvent[], ExemplarEvent[]>(annotations => {
```

Comment:
nit: nice for readability but not needed
```suggestion
          map(annotations => {
```